src/chk/chk_engine.c

```diff
@@ -1271,13 +1358,13 @@ chk_engine_notify(uint64_t gen, uuid_t uuid, d_rank_t rank, uint32_t phase,
 		if (unlikely(cbk->cb_phase >= phase))
 			D_GOTO(out, rc = -DER_NOTAPPLICABLE);
 
-		rc = chk_engine_setup_pools(ins, false);
-		if (rc == 0) {
-			ABT_mutex_lock(ins->ci_abt_mutex);
-			cbk->cb_phase = phase;
-			ABT_cond_broadcast(ins->ci_abt_cond);
-			ABT_mutex_unlock(ins->ci_abt_mutex);
-		}
+		ABT_mutex_lock(ins->ci_abt_mutex);
+		cbk->cb_phase = phase;
```

Comment:
For this case, it cannot. Because the former check guarantee that current engine check phase is smaller than the leader check phase. There is only one check leader, no others can make the check engine check phase to be higher at the moment.